server: Update docker image to bookworm and add CI workflow to test it

anlambert opened the pull request:
Following #1061, update the svix-server docker image to debian bookworm.

Add a CI workflow to test image build and container execution.

anlambert commented on .github/workflows/server-docker-image.yml:
```diff
@@ -0,0 +1,42 @@
+      working-directory: ./server
+
+    - name: Check server container is up and running
+      run: curl --retry 5 --retry-all-errors http://localhost:8071/api/v1/health/
```
`curl` uses exponential backoff when retrying so it waits 1 second after the first try, then 2, 4, 8, 16 and finally 32 seconds. It seems sufficient for the server to be up and API responsive.

tasn replied:
Yeah, 100%, great. :)

anlambert commented:
> use apt instead of apt-get and remove not required package version pins.

It seems the linting job does not like it, will update.

tasn commented:
No idea why it's failing again. :)

anlambert commented:
> No idea why it's failing again. :)

Because I removed the version pins on packages, sigh .. That linting rule is quite weird for debian (only the debian version should be pinned from my point of view) but anyway I will restore them. I removed other workflows on my dev branch when working on updating the docker file so that's why I did not spot these linting issues.

tasn commented:
I think version pinning is probably a good idea. It removes the mystery from Dockerfiles. E.g. my local testing my have a cached layer with an old zlib, and the release will be with a new zlib that may have issues. So I'm not against that. :)

anlambert commented:
Package version pins restored, all CI jobs should passed now.

tasn commented:
Awesome, thanks. Rerunning!

tasn commented:
New version is out based on the updated dockerfile, thanks again!
